platform/docs/docs/development/types.md

```diff
@@ -0,0 +1,127 @@
+---
+sidebar_position: 10
+sidebar_label: Global Types
+---
+
+# Extending App Types and Services in Your Application
+
+This documentation provides an overview and examples on how to use and extend `withAppTypes`, integrate custom properties, and add services in the global namespace of the application. This helps in enhancing the application's modularity and extensibility.
+
+## Overview of `withAppTypes`
+
+The `withAppTypes` function is a TypeScript utility that extends the base properties of components or modules with the application's core service and manager types. It allows for a more flexible and type-safe way to pass around core functionality and custom properties.
+
+### Using `withAppTypes`
+
+`withAppTypes` can be enhanced using generics to include custom properties. This is particularly useful for passing additional data or configurations specific to your component or service.
+
+#### Basic Syntax
+
+```typescript
+export type withAppTypes<T = object> = T & AppTypes.Services & AppTypes.Managers & {
+    [key: string]: any;
+};
+```
```

Comment:
user doesn't need to know the internal of the withAppTypes, usage is more important